microprofile/WFLY-19592_MicroProfile_Fault_Tolerance_4_1.adoc

```diff
@@ -0,0 +1,132 @@
+
+=== Related Issues
+
+* https://issues.redhat.com/browse/WFLY-19588[WFLY-19588 Implement MicroProfile Platform 7] – MP Platform parent issue
+* https://issues.redhat.com/browse/WFLY-19590[WFLY-19590 MicroProfile Telemetry 2.0] – dependency of this issue for full Telemetry metrics integration
+
+// EAP Tracking issue:
+// * https://issues.redhat.com/browse/EAP7-2225[EAP7-2225 MicroProfile Fault Tolerance 4.1]
+
+=== Affected Projects or Components
+
+* https://github.com/wildfly/wildfly[WildFly]
+
+=== Other Interested Projects
+
+* https://github.com/eclipse/microprofile-fault-tolerance[Eclipse MicroProfile Fault Tolerance] – MP Specification and TCK sources
+* https://github.com/smallrye/smallrye-fault-tolerance[SmallRye Fault Tolerance] – implementation
+
+=== Relevant Installation Types
+
+* Traditional standalone server (unzipped or provisioned by Galleon)
+* OpenShift Source-to-Image (S2I)
+* Bootable jar
+
+== Requirements
+
+* Upgrade MP FT specification modules to 4.1.
+* Upgrade SmallRye FT to 6.5.x.
+* Enable collecting metrics with MP Telemetry when the subsystem is available in the server configuration.
+
+=== Changed requirements
+
+* N/A
+
+=== Non-Requirements
+
+* N/A
+
+=== Future Work
+
+None.
+
+== Backwards Compatibility
+
+The specification and the implementation do not introduce any incompatibilities from the previous version (MP FT 4.0).
+
+=== Default Configuration
+
+There are no changes to the existing configuration.
+
+=== Importing Existing Configuration
+
+The subsystem does not have any configuration, so existing subsystem configuration is not affected.
+
+=== Deployments
+
+The specification and the implementation do not introduce any incompatibilities from the previous version relevant to deployments.
+
+=== Interoperability
+
+* N/A
+
+== Implementation Plan
+
+For the integration with Telemetry metrics to function,
+the integration of its https://issues.redhat.com/browse/WFLY-19590[subsystem and modules] is required.
+Implementation will be done in the created `mp-7-wip` https://github.com/wildfly/wildfly/tree/mp-7-wip[branch] in WildFly upstream repository.
+
+== Admin Clients
+
+* No effect on the existing clients, there are no model changes.
+
+== Security Considerations
+
+* No security implications of this integration.
+
+[[test_plan]]
+== Test Plan
+
+The following testsuites will be used to validate this change:
+
+* MicroProfile Fault Tolerance TCK tests will be added and executed by CI as part of the WildFly testsuite (in `testsuite/integration/microprofile-tck/fault-tolerance` module)
+* WildFly-specific behavior and configuration is covered by existing tests in the WildFly testsuite (in `testsuite/integration/microprofile` module)
+* Upstream https://github.com/smallrye/smallrye-fault-tolerance/tree/main/testsuite[implementation] testsuite is executed.
+* The corresponding MicroProfile tests from https://github.com/jboss-eap-qe/eap-microprofile-test-suite[EAP MicroProfile test suite]
```

Comment:
@rhusar Does this need something about testing the integration with MP Telemetry Metrics? Or is that covered by the TCK?

Reply:
I believe that's the aspect of the QE test plan not this one. But if @fabiobrz wants added something here just no problem, ideally using https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/incorporating-feedback-in-your-pull-request#applying-a-suggested-change

Reply:
Test details are for sure to be mentioned in the test plan, but I think that adding a simple note here could be beneficial, i.e. just to say that both the TCKs and the MicroProfile test suite include tests that verify the MP FT -> MP Telemetry 2.0 integration.

Reply:
Thanks @rhusar - BTW, I've added a couple of suggestions.